### ISSUE TITLE
Speed up upload & download speed

### ISSUE THREAD

**Issue by tulkarxxx** [Nekmo/telegram-upload#140]
* telegram-upload version:
* Python version:
* Operating System:
* Dependencies list (run `pip freeze`):

### Description

Upload speed is low, please implement "kotatogram's upload speed boost big"

### What I Did

```
Paste the command(s) you ran and the output.
If there was a crash, please include the traceback here.
```

**Comment by doublep21** [comment]
https://gist.github.com/painor/7e74de80ae0c819d3e9abcf9989a8dd6

**Comment by Nekmo** [comment]
I'll look into it, thanks!

**Comment by Aokromes** [comment]
looking for this feature :) i am trying to make backup of a lot of files and it takes a lot of time.

**Comment by nimma0001** [comment]
What's the progress on this really need this

**Comment by Nekmo** [comment]
This feature will be available in a future release.

**Comment by Nirzak** [comment]
Is this feature available yet? experiencing slow speed on telegram-upload

**Comment by tosankrisna** [comment]
is this feature already implemented?

**Comment by Nekmo** [comment]
This feature is not yet implemented. It will be implemented later.

**Comment by Nekmo** [comment]
This feature is available in testing in the branch issue-140:

```
pip install https://github.com/Nekmo/telegram-upload/archive/refs/heads/issue-140.zip
```

**Comment by xd003** [comment]
Just curious if you also implemented stuffs from here - https://gist.github.com/painor/7e74de80ae0c819d3e9abcf9989a8dd6 ?

**Comment by Nekmo** [comment]
@xd003 no, the solution is written from scratch

**Comment by Nekmo** [comment]
Fixed in version v0.7.0.

**Comment by tulkarxxx** [comment]
Hi, I am not able fo find https://github.com/Nekmo
/telegram-upload/archive/refs/heads/issue-140.zip

**Comment by Nekmo** [comment]
@tulkarxxx the branch is now merged into the master branch. You can test this new feature in the default branch of the repository:

```
pip install https://github.com/Nekmo/telegram-upload/archive/refs/heads/issue-140.zip
```

Tell me any bugs you find in the branch, thanks. This feature have a lot of changes in the source code.

**Comment by tulkarxxx** [comment]
HI, I get this error when trying to update:
Collecting
https://github.com/Nekmo/telegram-upload/archive/refs/heads/issue-140


.zip
  ERROR: HTTP error 404 while getting
https://github.com/Nekmo/telegram-upload/a


rchive/refs/heads/issue-140.zip
  ERROR: Could not install requirement
https://github.com/Nekmo/telegram-upload/


archive/refs/heads/issue-140.zip because of error 404 Client Error: Not
Found fo

              r url:
https://codeload.github.com/Nekmo/telegram-upload/zip/refs/heads/issue-14


  0
ERROR: Could not install requirement
https://github.com/Nekmo/telegram-upload/ar


chive/refs/heads/issue-140.zip because of HTTP error 404 Client Error: Not
Found

           for url:
https://codeload.github.com/Nekmo/telegram-upload/zip/refs/heads/issue


-140 for URL
https://github.com/Nekmo/telegram-upload/archive/refs/heads/issue-1


40.zip

**Comment by Nekmo** [comment]
Hi @tulkarxxx , 

Sorry for the typo, the correct url is:

```
pip install https://github.com/Nekmo/telegram-upload/archive/refs/heads/master.zip
```

Regards,

**Comment by Nekmo** [comment]
Speed improvements have been implemented in Telegram-upload v0.7.0. You can see the benchmarks with the improvement in:
https://docs.nekmo.org/telegram-upload/upload_benchmark.html
